pkg/controller/control/control_controller_test.go

```diff
@@ -131,30 +123,43 @@ func TestControlController(t *testing.T) {
 	t.Run("should create secret for control certificates", func(t *testing.T) {
 		secret := &core.Secret{}
 		err = Cl.Get(context.Background(), types.NamespacedName{
-			Name:      "test-control-secret-certificates",
-			Namespace: "default",
+			Name:      controlName.Name + "-secret-certificates",
+			Namespace: controlName.Namespace,
 		}, secret)
 		assert.NoError(t, err)
 		assert.NotEmpty(t, secret)
 		expectedOwnerRefs := []v1.OwnerReference{{
-			APIVersion: "contrail.juniper.net/v1alpha1", Kind: "Control", Name: "test-control",
+			APIVersion: "contrail.juniper.net/v1alpha1", Kind: "Control", Name: controlName.Name,
 			Controller: &trueVal, BlockOwnerDeletion: &trueVal,
 		}}
 		assert.Equal(t, expectedOwnerRefs, secret.OwnerReferences)
 	})
 
-	t.Run("should create and check for StatefulSet", func(t *testing.T) {
-		sts := &contrail.Control{}
+	t.Run("should create control resource", func(t *testing.T) {
+		control := &contrail.Control{}
 		falseVal := false
-		err = Cl.Get(context.Background(), types.NamespacedName{
-			Name:      "test-control",
-			Namespace: "default",
-		}, sts)
+		err = Cl.Get(context.Background(), controlName, control)
 		assert.NoError(t, err)
 		expectedStatus := contrail.ControlStatus{Active: &falseVal}
 		require.NotNil(t, expectedStatus.Active, "expectedStatus.Active should not be nil")
-		require.NotNil(t, sts.Status.Active, "sts.Status.Active Should not be nil")
-		assert.Equal(t, *expectedStatus.Active, *sts.Status.Active)
+		require.NotNil(t, control.Status.Active, "sts.Status.Active Should not be nil")
+		assert.Equal(t, *expectedStatus.Active, *control.Status.Active)
 	})
 
+	t.Run("should create StatefulSet with dataSubnet annotation in pod template", func(t *testing.T) {
+		sts := &apps.StatefulSet{}
+		err = Cl.Get(context.Background(), types.NamespacedName{
+			Name:      controlName.Name + "-control-statefulset",
+			Namespace: controlName.Namespace,
+		}, sts)
+		assert.NoError(t, err)
+		expectedAnnotation := map[string]string{"dataSubnet": "172.17.90.0/24"}
+		assert.Truef(
+			t,
+			reflect.DeepEqual(sts.Spec.Template.Annotations, expectedAnnotation),
+			"expected annotation not applied, expected: %v, got: %v",
+			expectedAnnotation,
+			sts.Spec.Template.Annotations,
```

Comment:
assert.Equal is not enough for this check?

Reply:
it is enough, it already use DeepEqual underneath, thank you for pointing that out